runtime/node/src/shared/helpers.ts

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import fs from 'fs';
+import minimist from 'minimist';
+import path from 'path';
+import { Server } from 'restify';
+import merge from 'lodash/merge';
+import {
+  BotFrameworkAdapter,
+  BotFrameworkAdapterSettings,
+  ChannelServiceRoutes,
+  ConversationState,
+  InputHints,
+  SkillHandler,
+  TurnContext,
+  UserState,
+  WebRequest,
+  WebResponse,
+} from 'botbuilder';
+import { AuthenticationConfiguration, SimpleCredentialProvider } from 'botframework-connector';
+import { ComposerBot } from './composerBot';
+import { BotSettings } from './settings';
+import { SkillConversationIdFactory } from './skillConversationIdFactory';
+
+/**
+ * Get listening port for listening from environment variables or arguments.
+ */
+export const getServerPort = () => {
+  const argv = minimist(process.argv.slice(2));
+  // prefer the argv port --port=XXXX over process.env because the parent Composer app uses that.
+  const port = argv.port || process.env.port || process.env.PORT || 3979;
+  return port;
+};
+
+/**
+ * Get path of bot project.
+ */
+export const getProjectRoot = (): string => {
+  // get the root folder according to environment
+  if (process.env.node_environment === 'production') {
```

Comment:
use `NODE_ENV` instead.

Reply:
the NODE_ENV will be set to production when yarn start, but we need to getProjectRoot specialize by local run or deploy. So we use this new environment instead.

Reply:
changed

Reply:
NODE_ENV is set to production for what?

I am with @a-b-r-o-w-n here. NODE_ENV should not just be set to production by default. If this is an environment inherited from the Composer app, we should find a way around this but I don't think introducing a new name is the right fix.